File: br_frontend/src/app/models/product.ts

export class Product{
    id:number;
    sku:string;
    imageUrl: string[]
    name:string;
    description:string;
    stockQuantity:number;
    categoryName:string;
    originalPrice:number;
    discountPercentage:number;
    category:string;
    rating:number;

    constructor(
      id:number,
      sku:string,
      imageUrl: string[],
      name:string,
      description:string,
      stockQuantity:number,
      categoryName:string,
      originalPrice:number,
      discountPercentage:number,
      category:string,
      rating:number
      ) {
        this.id = id;
        this.name = name;
        this.imageUrl = imageUrl;
        this.originalPrice = originalPrice;
        this.discountPercentage = discountPercentage;
        this.description = description;
        this.category = category;
        this.rating = rating;
        this.sku = sku;
        this.stockQuantity = stockQuantity;
        this.categoryName = categoryName;
      }
    
    getDiscountedPrice(): number {
        return this.originalPrice - (this.originalPrice * this.discountPercentage) / 100;
      }
}

File: br_backend/product_service/src/main/java/com/example/product_service/models/Product.java
package com.example.product_service.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Product {
    String id;
    String name;
    String imageUrl[];
    int originalPrice;
    int discountPercentage;
    String description;
    String category;
    float Rating;
}
